src/metatensor/models/cli/train_model.py

```diff
@@ -336,13 +341,43 @@ def _train_model_hydra(options: DictConfig) -> None:
         device_str=options["device"],
     )
 
-    save_model(model, options["output_path"])
+    save_model(model, f'{options["output_path"][:-3]}.ckpt')
+    export(model, options["output_path"])
+
+    logger.info("Setting up evaluation")
+    exported_model = torch.jit.load(options["output_path"])
+    requested_neighbors_lists = exported_model.requested_neighbors_lists()
+
+    # working around https://github.com/lab-cosmo/metatensor/issues/521
+    # Desired:
+    # for structure, _ in train_dataset:
+    #     attach_neighbor_lists(structure, requested_neighbors_lists)
+    # for structure, _ in validation_dataset:
+    #     attach_neighbor_lists(structure, requested_neighbors_lists)
+    # for structure, _ in test_dataset:
+    #     attach_neighbor_lists(structure, requested_neighbors_lists)
+    # Current:
+    train_dataloader = torch.utils.data.DataLoader(
```

Comment:
I actually love this workaround a lot, since it makes the model-specific training script cleaner. However, aren't we getting into a double NL calculation step with rascaline-based models?

Reply:
No, in evaluation mode rascaline asks for a neighbor list from outside. During training mode, it calculates it itself

Reply:
to be more precise: rascaline checks if neighbors lists are already present, and if not compute them itself. So if you provide NL in training mode, they will be used!

Reply:
It actually errors at evaluation time when wrapped inside MetatensorAtomisticModel, so we have to provide neighbor lists at evaluation time. Of course it's not a rascaline check though